### PR TITLE
refactor(ingest): add dynamic feed parsers lookup

### PR DESCRIPTION
**Pull request by thnkloud9**

### CODE REVIEW

File: scripts/apps/ingest/services/IngestProviderService.js [hunk 1]
```diff
@@ -16,6 +16,36 @@ export function IngestProviderService(api, $q, preferencesService, $filter, sear
...
+
+                if (result._links.next) {
+                    pg++;
+                    return _getAllFeedParsersAllowed(criteria, pg, merged);
+                }
+                return $filter('sortByLabel')(merged);
+            });
+    };
+
+    // TODO: not implemented yet.  should be used when dynamic services with custom
```
Comment (gbbr) [inline scripts/apps/ingest/services/IngestProviderService.js]: It's a bit unclear what is not implemented... The function seems to have code, so it is implemented. Or do you mean the API endpoint? If so, maybe the comment belongs above the line `api.query` and might be worth specifying that the backend is unimplemented. And if that's the case, then maybe it's not even worth having this comment here?

**Comment by gbbr** [comment]
This PR contains very little changes and it spans across 3 different commits. Please squash when merging.

File: scripts/apps/ingest/services/IngestProviderService.js (hunk 1 above)
Comment (thnkloud9) [inline scripts/apps/ingest/services/IngestProviderService.js]: yeah, probably you are right, better to remove the comment.

File: scripts/apps/ingest/tests/sdIngestSourcesContent.spec.js [hunk 2]
```diff
@@ -28,7 +29,7 @@ describe('sdIngestSourcesContent directive', () => {
         scope.$digest();
     }));
 
-    it('initializes the list of available field names in scope', () => {
+    fit('initializes the list of available field names in scope', () => {
```
Comment (akintolga) [inline scripts/apps/ingest/tests/sdIngestSourcesContent.spec.js]: `fit` is left here

File: scripts/core/filters/index.js [hunk 3]
```diff
@@ -198,6 +198,11 @@ export default angular.module('superdesk.core.filters', [])
             _entry[propertyName].toLowerCase()
             : _entry.name.toLowerCase());
     })
+    .filter('sortByLabel', () => function(_collection, propertyName = 'label') {
```
Comment (mdhaman) [inline scripts/core/filters/index.js]: There is already `filter('sortByName')` and examples are:
 https://github.com/superdesk/superdesk-client-core/blob/master/scripts/apps/vocabularies/services/VocabularyService.js#L54 or https://github.com/superdesk/superdesk-client-core/blob/master/scripts/apps/templates/directives/TemplatesDirective.js#L20

Comment (thnkloud9) [inline scripts/core/filters/index.js]: good call, removing duplicate.

**Comment by thnkloud9** [comment]
refactor(ingest): add dynamic feed parsers lookup

**Pull request merged**